app/scripts/components/analysis/define/aoi-selector.tsx

```diff
@@ -0,0 +1,155 @@
+import React, { useEffect, useRef } from 'react';
+import styled from 'styled-components';
+import { Feature, MultiPolygon, Polygon } from 'geojson';
+import { useDeepCompareMemo } from 'use-deep-compare';
+import bbox from '@turf/bbox';
+
+import {
+  Toolbar,
+  ToolbarIconButton,
+  ToolbarLabel,
+  VerticalDivider
+} from '@devseed-ui/toolbar';
+import {
+  Dropdown,
+  DropMenu,
+  DropMenuItem,
+  DropTitle
+} from '@devseed-ui/dropdown';
+import {
+  CollecticonArea,
+  CollecticonEllipsisVertical,
+  CollecticonTrashBin,
+  CollecticonUpload2
+} from '@devseed-ui/collecticons';
+
+import {
+  Fold,
+  FoldHeader,
+  FoldHeadline,
+  FoldHeadActions,
+  FoldTitle,
+  FoldBody
+} from '$components/common/fold';
+import MapboxMap, { MapboxMapRef } from '$components/common/mapbox';
+import {
+  AoiChangeListenerOverload,
+  AoiState
+} from '$components/common/aoi/types';
+
+const MapContainer = styled.div`
+  position: relative;
+`;
+
+const DemoMap = styled(MapboxMap)`
+  min-height: 24rem;
+`;
+
+interface AoiSelectorProps {
+  qsFeature: Feature<MultiPolygon> | null;
+  aoiDrawState: AoiState;
+  onAoiEvent: AoiChangeListenerOverload;
+}
+
+export default function AoiSelector(props: AoiSelectorProps) {
+  const { onAoiEvent, qsFeature, aoiDrawState } = props;
+  const { selected, drawing, feature } = aoiDrawState;
+  const mapRef = useRef<MapboxMapRef>();
+
+  // Technical debt
+  // Despite the query parameters support for multiple features on the aoi, the
+  // AOI drawing tool only supports one.
+  // Keeping just the first one.
+  const polygon: Feature<Polygon> | null = useDeepCompareMemo(() => {
```

Comment:
That is true, but it would involve passing the raw string around, which I think it is confusing.
Nevertheless, qsFeature _should_ be a stable object (or possible to be made one), but I need to double check.